xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -764,28 +763,34 @@
       #set ($matchType = 'partial')
     #end
   #end
-  #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType)")
+  #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType false $multilistsize)")
   #set ($whereSql = "${whereSql} and $filterQuery.trim()")
-  #livetable_addFilterParam($filterValue $matchType $whereParams)
+  #livetable_addFilterParam($filterValue $matchType $whereParams $multilistsize)
 #end
 
 
-#macro (livetable_getFilterQuery $column $matchType $isList)
+#macro (livetable_getFilterQuery $column $matchType $isList $multilistsize)
```

Comment:
I would rename ``$multilistsize`` to ``$valueCount`` (or something similar, camel case anyway) in order to make it more generic. For instance, when ``$valueCount`` is ``1``

    $column in (?)

is equivalent to

    $column = ?

And in the future we may want to add support for matching against multiple values even for the *partial* and *prefix* matching type

    ($stringtool.repeat("upper($column) like upper(?)", ' or ', $valueCount))